zip.go

```diff
@@ -6,11 +6,26 @@ import (
 	"compress/flate"
 	"fmt"
 	"io"
+	"io/ioutil"
 	"log"
 	"os"
 	"path"
 	"path/filepath"
 	"strings"
+
+	"github.com/dsnet/compress/bzip2"
+	"github.com/klauspost/compress/zstd"
+	"github.com/ulikunitz/xz/lzma"
+)
+
+// Compression methods.
+const (
+	Store      uint16 = 0  // no compression
+	Deflate    uint16 = 8  // DEFLATE compressed
+	BZIP2      uint16 = 12 // bzip2
+	LZMA       uint16 = 14 //LZMA
+	ZSTD       uint16 = 20 //see https://pkware.cachefly.net/webdocs/casestudies/APPNOTE.TXT.
+	WINZIPZSTD uint16 = 93
```

Comment:
This package is shared by many types, so these constant names should probably be prefixed by something like `Zip` since they're specific to the zip archiver.

They should also be a custom type perhaps, not just "any" `uint16`, e.g. `type ZipCompressionMethod uint16`.

Also clean up the comments (spacing).